### ISSUE TITLE
[Symfony auto-instrumentation] Messenger instrumentation should use SpanKind::KIND_PRODUCER, SpanKind::KIND_CONSUMER

### ISSUE THREAD

**Issue by dkarlovi** [open-telemetry/opentelemetry-php#1314]
**Steps to reproduce**

Enable Symfony auto-instrumentation with default setup.

**What is the expected behavior?**

Spans when creating the message should be marked with `SpanKind::KIND_PRODUCER`.

Spans when **handling** the message should be marked with `SpanKind::KIND_CONSUMER`.

https://opentelemetry.io/docs/specs/semconv/messaging/messaging-spans/#span-kind

**What is the actual behavior?**

Everything is marked `KIND_INTERNAL` and it seems the consumer is currently not instrumented at all (I might be wrong here, please feel free to correct me).

Additionally, it seems to me the names of the spans are also not correct, as shown in the table in the spec linked above?

**Comment by AA2512** [comment]
@agoallikmaa 
I can take this up

**Comment by RichardChukwu** [comment]
Hello @AA2512  please I'd like to know if this is still being worked please? @brettmc for possible assigning

Thank you

**Comment by brettmc** [comment]
Hello and welcome, @RichardChukwu 
Please feel free to submit to work on this and submit a PR.

**Comment by RichardChukwu** [comment]
Got it, I'll do just that. Thank you @brettmc

**Comment by chiomalovet** [comment]
Hi @brettmc  can i take this up

**Comment by dkarlovi** [comment]
@chiomalovet it was just assigned to @RichardChukwu, unless he changed his mind, he should get dibs for a while to give it a shot.

**Comment by dkarlovi** [comment]
> Let me look for another one to work on

Great, there's not a shortage of work on the project(s)! :smile:

**Comment by RichardChukwu** [comment]
Hello @dkarlovi please can you specify the exact folder location for this. I can't seem to figure out here symfony auto-instrumentation is in the repo

**Comment by dkarlovi** [comment]
@RichardChukwu I know what you mean, I have the same issue navigating these OTEL repos, it's very confusing if you're not here day to day, I was about to suggest the contrib repo gets merged into this repo since the packages are now released from the split repos anyway. WDYT @bobstrecansky @brettmc?

To answer your question, it's here I believe:
https://github.com/open-telemetry/opentelemetry-php-contrib/tree/main/src/Instrumentation/Symfony

**Comment by RichardChukwu** [comment]
Hello @brettmc kindly review PR #304 as a fix I submitted to this issue. I'll appreciate feedback where necessary too. Thank you

**Comment by brettmc** [comment]
@RichardChukwu some additional changes noted above, if you're interested in working on them :)

**Comment by RichardChukwu** [comment]
Before working on this, just want to ensure i get you correctly @dkarlovi 

Do you mean updating the code to introduce new spans with KIND_CONSUMER for the parts of the code handling message reception or processing as it doesn't exist?

For instance, based on the spec from what I can see, I presume key changes will be:


- Adding the "consume" operation with SpanKind::KIND_CONSUMER for receiving messages.

- Updating span names to use publish, send, and consume as per the spec, ensuring naming consistency for operations.

- Instrumenting the SenderInterface for both sending and receiving spans, with the appropriate SpanKind (producer for sending and consumer for receiving).

Your thoughts please....

**Comment by dkarlovi** [comment]
> Do you mean updating the code to introduce new spans with KIND_CONSUMER for the parts of the code handling message reception or processing as it doesn't exist?

Yes, correct. The messages being created and sent is currently fully instrumented, but them being received and processed is not instrumented at all, from my understanding at least.

**Comment by RichardChukwu** [comment]
Alright, got it. I'll work on that and send a PR for a fix soon @dkarlovi
